SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -12901,81 +12917,106 @@ void CEulerSolver::BC_Fluid_Interface(CGeometry *geometry, CSolver **solver_cont
+              P_static   = PrimVar_j[nDim+1];
+              rho_static = PrimVar_j[nDim+2];           
+              FluidModel->SetTDState_Prho(P_static, rho_static);
 
-            Secondary_j[0] = FluidModel->GetdPdrho_e();
-            Secondary_j[1] = FluidModel->GetdPde_rho();  
+              Secondary_j[0] = FluidModel->GetdPdrho_e();
+              Secondary_j[1] = FluidModel->GetdPde_rho();  
 
-            conv_numerics->SetSecondary(Secondary_i, Secondary_j);
-          }
+              conv_numerics->SetSecondary(Secondary_i, Secondary_j);
+            }
 
-          /*--- Set the normal vector ---*/
+            /*--- Set the normal vector ---*/
+ 
+            geometry->vertex[iMarker][iVertex]->GetNormal(Normal);
+            for (iDim = 0; iDim < nDim; iDim++) 
+              Normal[iDim] = -Normal[iDim];
 
-          geometry->vertex[iMarker][iVertex]->GetNormal(Normal);
-          for (iDim = 0; iDim < nDim; iDim++) 
-            Normal[iDim] = -Normal[iDim];
+            conv_numerics->SetNormal(Normal);
 
-          conv_numerics->SetNormal(Normal);
+            if (grid_movement)
+              conv_numerics->SetGridVel(geometry->node[iPoint]->GetGridVel(), geometry->node[iPoint]->GetGridVel());
 
-          if (grid_movement)
-            conv_numerics->SetGridVel(geometry->node[iPoint]->GetGridVel(), geometry->node[iPoint]->GetGridVel());
+            /*--- Compute the convective residual using an upwind scheme ---*/
 
-          /*--- Compute the convective residual using an upwind scheme ---*/
+            conv_numerics->ComputeResidual(tmp_residual, Jacobian_i, Jacobian_j, config);
 
-          conv_numerics->ComputeResidual(Residual, Jacobian_i, Jacobian_j, config);
+            for (iVar = 0; iVar < nVar; iVar++)
+              Residual[iVar] += coeff*tmp_residual[iVar];
 
-          /*--- Add Residuals and Jacobians ---*/
+          }
 
+          /*--- Add Residuals and Jacobians ---*/
+  
           LinSysRes.AddBlock(iPoint, Residual);
           if (implicit) 
             Jacobian.AddBlock(iPoint, iPoint, Jacobian_i);
 
           if (viscous) {
+            
+            for (iVar = 0; iVar < nVar; iVar++)
+              Residual[iVar] = 0.0;
+            
+            for (jVertex = 0; jVertex < nDonorVertex; jVertex++){
+              PrimVar_j[nDim+5] = GetSlidingState(iMarker, iVertex, nDim+5, jVertex); 
+              PrimVar_j[nDim+6] = GetSlidingState(iMarker, iVertex, nDim+6, jVertex); 
 
-            PrimVar_j[nDim+5] = GetSlidingState(iMarker, iVertex, nDim+5);
-            PrimVar_j[nDim+6] = GetSlidingState(iMarker, iVertex, nDim+6);
-
-            /*--- Set the normal vector and the coordinates ---*/
-
-            visc_numerics->SetNormal(Normal);
-            visc_numerics->SetCoord(geometry->node[iPoint]->GetCoord(), geometry->node[Point_Normal]->GetCoord());
+              coeff = GetSlidingState(iMarker, iVertex, nPrimVar, jVertex);
```

Comment:
Maybe, can you add a comment about "coeff", adding why for more then one vertex is needed? What about a different name?